src/simplify_qdq.cpp

```diff
@@ -442,10 +572,10 @@ void remove_qdq_pairs(module& m)
         {
             if(arg->name() == "dequantizelinear")
             {
-                auto q = arg->inputs().front();
-                if((q->name() == "quantizelinear") and is_same_scale_zero(arg, q))
+                auto curr = skip_dq_inputs_to_q(arg->inputs().front());
```

Comment:
Can you just rewrite this to use matchers instead of a raw loop? It seems much simpler than writing another version of matchers here.